Pcap++/src/RawSocketDevice.cpp

```diff
@@ -216,26 +217,34 @@ RawSocketDevice::RecvPacketResult RawSocketDevice::receivePacket(RawPacket& rawP
 #endif
 }
 
-int RawSocketDevice::receivePackets(RawPacketVector& packetVec, int timeout, int& failedRecv)
+int RawSocketDevice::receivePackets(RawPacketVector& packetVec, double timeout, int& failedRecv)
 {
 	if (!isOpened())
 	{
 		PCPP_LOG_ERROR("Device is not open");
 		return 0;
 	}
 
-	long curSec, curNsec;
-	clockGetTime(curSec, curNsec);
+	int64_t timeoutMilli = timeout * 1000;
 
 	int packetCount = 0;
 	failedRecv = 0;
 
-	long timeoutSec = curSec + timeout;
+	auto start = std::chrono::system_clock::now();
```

Comment:
I recommend using `steady_clock` or `high_resolution_clock` as `system_clock` may not be steady and cause inaccuracies when used to calculate a wait duration.

Reply:
Changed. Thanks.